Delete Cocoa specific WebExtensionSQLite implementation

ItsJamie9494 opened the pull request:
#### 85912f336e74a1ce70c668c2ea94d047d06ae648
<pre>
Delete Cocoa specific WebExtensionSQLite implementation
<a href="https://bugs.webkit.org/show_bug.cgi?id=301312">https://bugs.webkit.org/show_bug.cgi?id=301312</a>

Reviewed by Timothy Hatcher.

All SQLite Databases have been ported to C++, so the old Cocoa implementations are completely unused, this therefore deletes those classes.

No tests are needed, we are just removing unused code and moving existing code around.

* Source/WebKit/Shared/Extensions/_WKWebExtensionSQLiteDatabase.h: Removed.
* Source/WebKit/Shared/Extensions/_WKWebExtensionSQLiteDatabase.mm: Removed.
* Source/WebKit/Shared/Extensions/_WKWebExtensionSQLiteDatatypeTraits.h: Removed.
* Source/WebKit/Shared/Extensions/_WKWebExtensionSQLiteHelpers.h: Removed.
* Source/WebKit/Shared/Extensions/_WKWebExtensionSQLiteRow.h: Removed.
* Source/WebKit/Shared/Extensions/_WKWebExtensionSQLiteRow.mm: Removed.
* Source/WebKit/Shared/Extensions/_WKWebExtensionSQLiteStatement.h: Removed.
* Source/WebKit/Shared/Extensions/_WKWebExtensionSQLiteStatement.mm: Removed.
* Source/WebKit/Shared/Extensions/_WKWebExtensionSQLiteStore.h: Removed.
* Source/WebKit/Shared/Extensions/_WKWebExtensionSQLiteStore.mm: Removed.
* Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIDeclarativeNetRequestCocoa.mm:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/302044@main">https://commits.webkit.org/302044@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4151d06b04cc515876b37523296084472b831fe1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/127816 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/47464 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/38634 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/135193 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/79375 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/48096 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/55996 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/97300 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/79375 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/130764 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/38466 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/114492 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/77870 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/37257 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/78498 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/108350 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/33057 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/137671 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/54475 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/42010 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/105830 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/54987 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/110847 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/105563 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/50991 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/29427 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/52114 "The change is no longer eligible for processing.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19982 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/54413 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/60905 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [⏳ 🛠 jsc-armv7 ](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/53649 "Built successfully") | | [⏳ 🧪 jsc-armv7-tests ](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [⏳ 🛠 watch ](https://ews-build.webkit.org/#/builders/watchOS-11-Build-EWS "Waiting in queue, processing has not started yet") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/55405 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

xeenon reviewed (approved):
Should remove the part about StorageSQLite in the commit message.

ItsJamie9494 committed:
Delete Cocoa specific WebExtensionSQLite implementation
https://bugs.webkit.org/show_bug.cgi?id=301312

Reviewed by Timothy Hatcher.

All SQLite Databases have been ported to C++, so the old Cocoa implementations are completely unused, this therefore deletes those classes.

No tests are needed, we are just removing unused code and moving existing code around.

* Source/WebKit/Shared/Extensions/_WKWebExtensionSQLiteDatabase.h: Removed.
* Source/WebKit/Shared/Extensions/_WKWebExtensionSQLiteDatabase.mm: Removed.
* Source/WebKit/Shared/Extensions/_WKWebExtensionSQLiteDatatypeTraits.h: Removed.
* Source/WebKit/Shared/Extensions/_WKWebExtensionSQLiteHelpers.h: Removed.
* Source/WebKit/Shared/Extensions/_WKWebExtensionSQLiteRow.h: Removed.
* Source/WebKit/Shared/Extensions/_WKWebExtensionSQLiteRow.mm: Removed.
* Source/WebKit/Shared/Extensions/_WKWebExtensionSQLiteStatement.h: Removed.
* Source/WebKit/Shared/Extensions/_WKWebExtensionSQLiteStatement.mm: Removed.
* Source/WebKit/Shared/Extensions/_WKWebExtensionSQLiteStore.h: Removed.
* Source/WebKit/Shared/Extensions/_WKWebExtensionSQLiteStore.mm: Removed.
* Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIDeclarativeNetRequestCocoa.mm:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/302044@main
